test/sql/hnsw_create.sql

```diff
@@ -78,3 +79,30 @@ UPDATE small_world4 SET vector = '{0,0,0}' WHERE id = '001';
 \set ON_ERROR_STOP off
 CREATE INDEX ON small_world4 USING hnsw (vector) WITH (M=14, ef=22, ef_construction=2);
 \set ON_ERROR_STOP on
+
+-- Test index creation on empty table and no dimension specified
+CREATE TABLE small_world5 (
+    id SERIAL PRIMARY KEY,
+    v REAL[]
+);
+
+-- We can still create an index despite having an empty table and not specifying a dimension during index creation
+CREATE INDEX small_world5_hnsw_idx ON small_world5 USING hnsw (v dist_l2sq_ops);
+
+begin;
+-- Our index then infers the dimension from the first inserted row
+INSERT INTO small_world5 (id, v) VALUES
+('000', '{1,0,0,0,1}'),
```

Comment:
Can you insert a `NULL` value here first, before inserting anything else, to trigger [this case](https://github.com/lanterndata/lantern/pull/251/files#diff-757fbd016cf447a1d7cb23e47b01cd1d57a1e9f8325084e30d6d9564c7fdc01bR91) ?